File: app/src/main/java/com/example/testproguard/test6/base/presenter/BaseMvpPresenter.java
package com.example.testproguard.test6.base.presenter;

import android.content.Context;
import android.os.Bundle;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.example.testproguard.test6.base.view.IBaseMvpView;
import com.example.mylibrary.utils.Utils;

/**
 * MVP公共Presenter
 */
public class BaseMvpPresenter<V extends IBaseMvpView>  {

    private static final String TAG = BaseMvpPresenter.class.getSimpleName();

    protected V mView;

    public BaseMvpPresenter(V view) {
        this.mView = view;
    }

    public void showToast(String str) {
        if(mView != null)
            mView.showToast(str);
    }

    public Context getContext() {
        return mView != null ? mView.getContext() : null;
    }

    public V getView() {
        return mView;
    }

    public void onCreate(Bundle savedInstanceState) {
        Utils.i(TAG, "onCreate");
    }

    public void onStart() {
        Utils.i(TAG, "onStart");
    }

    public void onRestart() {
        Utils.i(TAG, "onRestart");
    }

    public void onPostResume() {
        Utils.i(TAG, "onPostResume");
    }

    public void onResume() {
        Utils.i(TAG, "onResume");
    }

    public void onPause() {
        Utils.i(TAG, "onPause");
    }

    public void onStop() {
        Utils.i(TAG, "onStop");
    }

    public void onDestroy() {
        Utils.i(TAG, "onDestroy");
    }

    public void onAttach(@NonNull Context context) {
        Utils.i(TAG, "onAttach");
    }

    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        Utils.i(TAG, "onAttach");
    }

    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        Utils.i(TAG, "onAttach");
    }

    public void onDestroyView() {
        Utils.i(TAG, "onAttach");
    }

    public void onDetach() {
        Utils.i(TAG, "onAttach");
    }

}

File: app/src/main/java/com/example/testproguard/test6/base/presenter/BaseMvpPresenter.kt
package com.example.testproguard.test6.base.presenter

import android.content.Context
import android.os.Bundle
import android.view.View
import com.example.mylibrary.utils.Utils
import com.example.testproguard.test6.base.view.IBaseMvpView

/**
 * MVP公共Presenter
 */
open class BaseMvpPresenter<V : IBaseMvpView?>(view: V) {
    var view: V?
        protected set

    init {
        this.view = view
    }

    fun showToast(str: String?) {
        if (view != null) view!!.showToast(str)
    }

    val context: Context?
        get() = if (view != null) view!!.context else null

    open fun onCreate(savedInstanceState: Bundle?) {
        Utils.i(TAG, "onCreate")
    }

    open fun onStart() {
        Utils.i(TAG, "onStart")
    }

    fun onRestart() {
        Utils.i(TAG, "onRestart")
    }

    fun onPostResume() {
        Utils.i(TAG, "onPostResume")
    }

    open fun onResume() {
        Utils.i(TAG, "onResume")
    }

    open fun onPause() {
        Utils.i(TAG, "onPause")
    }

    open fun onStop() {
        Utils.i(TAG, "onStop")
    }

    open fun onDestroy() {
        Utils.i(TAG, "onDestroy")
    }

    open fun onAttach(context: Context) {
        Utils.i(TAG, "onAttach")
    }

    open fun onActivityCreated(savedInstanceState: Bundle?) {
        Utils.i(TAG, "onAttach")
    }

    open fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        Utils.i(TAG, "onAttach")
    }

    open fun onDestroyView() {
        Utils.i(TAG, "onAttach")
    }

    open fun onDetach() {
        Utils.i(TAG, "onAttach")
    }

    companion object {
        private val TAG: String = BaseMvpPresenter::class.java.simpleName
    }
}
